components/NewDialog.tsx

```diff
@@ -0,0 +1,49 @@
+import { observer } from "mobx-react-lite";
+import { TodoStore } from "../stores/Todo.store";
+import { useState } from "react";
+import { Button, Dialog, Input, TextArea, VisuallyHidden } from "tamagui";
+import { NewTodoButton } from "./NewButton";
+
+type Props = { store: TodoStore };
+export const NewDialog = observer(function NewDialog({ store }: Props) {
+  const [isOpen, setOpen] = useState(false);
+  const [title, setTitle] = useState("");
+  const [text, setText] = useState("");
+
+  return (
+    <Dialog open={isOpen} defaultOpen={false}>
+      <Dialog.Trigger asChild>
+        <NewTodoButton onPress={() => setOpen(true)} />
+      </Dialog.Trigger>
+      <Dialog.Portal>
+        <Dialog.Content>
+          <Dialog.Title>Create a new to-do!</Dialog.Title>
+          <VisuallyHidden>
+            <Dialog.Description>
+              Enter your text to create a new item to-do.
+            </Dialog.Description>
+          </VisuallyHidden>
+
+          <Input onChangeText={setTitle} />
+          <TextArea onChangeText={setText} />
+
+          <Dialog.Close asChild>
+            <Button
+              onPress={() => {
+                const todo = {
+                  title: title,
+                  text: text,
+                };
+
+                store.addTodo(todo);
```

Comment:
instead passing the 'Store' to this component. You could pass a 'callback' like 'OnSave: (todo: ToDo) => void.

This way your NewDialog becomes fully re-usable and its not tied to the specific store.

Reply:
I've already done this in the next PR, to add editing todos.